File: VB/SecuredExportExample.Module/SecurityObjects/ExportPermissionData.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation
Imports DevExpress.ExpressApp.Security
Imports System.Collections.Generic

Namespace SecuredExportExample.Module.SecurityObjects
    <System.ComponentModel.DisplayName("ExportOperationName Permission")> _
    Public Class ExportPermissionData
        Inherits PermissionData
        Private _canExport As Boolean = True
        Protected Overrides Function GetPermissionInfoCaption() As String
            Return "ExportOperationName"
        End Function
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Overrides Function GetPermissions() As IList(Of IOperationPermission)
            Dim result As IList(Of IOperationPermission) = New List(Of IOperationPermission)()
            If _canExport Then
                result.Add(New ExportPermission())
            End If
            Return result
        End Function
        Public Property CanExport() As Boolean
            Get
                Return _canExport
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue("CanExport", _canExport, value)
            End Set
        End Property
    End Class

End Namespace

File: CS/SecuredExportExample.Module/SecurityObjects/ExportPermissionData.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.Security;
using System.Collections.Generic;

namespace SecuredExportExample.Module.SecurityObjects {
    [System.ComponentModel.DisplayName("ExportOperationName Permission")]
    public class ExportPermissionData : PermissionData {
        private bool canExport = true;
        protected override string GetPermissionInfoCaption() {
            return "ExportOperationName";
        }
        public ExportPermissionData(Session session)
            : base(session) {
        }
        public override IList<IOperationPermission> GetPermissions() {
            IList<IOperationPermission> result = new List<IOperationPermission>();
            if (canExport) {
                result.Add(new ExportPermission());
            }
            return result;
        }
        public bool CanExport {
            get { return canExport; }
            set { SetPropertyValue("CanExport", ref canExport, value); }
        }
    }

}
